src/Ipn/Handler.php

```diff
@@ -162,29 +162,30 @@ protected function getGroupType(array $ipn_data) {
    *   If the payment entity could not be found e.g. no longer exists.
    */
   protected function getPaymentEntity(array $ipn_data) {
-    if (empty($ipn_data['merchantTransactionId'])) {
+    if (empty($ipn_data['referenceNumber'])) {
       $message = sprintf(
-        'Cannot determine the payment ID for IPN of type "%s".',
+        'Cannot determine the transaction ID for IPN of type "%s".',
         $this->getType($ipn_data)
       );
       $this->logger->error($message);
       throw new BadRequestHttpException('Transaction ID required.');
     }
 
-    $payment = $this->entityTypeManager
-      ->getStorage('commerce_payment')
-      ->load($ipn_data['merchantTransactionId']);
-    if (!$payment) {
+    $payment_storage = $this->entityTypeManager->getStorage('commerce_payment');
+    $payment_id = $payment_storage->getQuery()
```

Comment:
@krystalcode Actually, `$payment_storage` has a function called loadByRemoteId(), so we can just do:

$payment_id = $payment_storage->loadByRemoteId($ipn_data['referenceNumber']);